testsuite/tests/typing-gadts/pr10189.ml

```diff
@@ -111,24 +111,6 @@ Warning 8 [partial-match]: this pattern-matching is not exhaustive.
 Here is an example of a case that is not matched:
 Some A
 val g : 'a M.j t option -> unit = <fun>
-|}, Principal{|
-module M :
-  sig
-    type 'a d
-    type i = < m : 'c. 'c -> 'c d >
-    type 'a j = < m : 'c. 'c -> 'a >
-  end
-type _ t = A : M.i t
-File "_none_", line 1:
-Warning 18 [not-principal]: typing this pattern requires considering $0 and 'c M.d as equal.
-But the knowledge of these types is not principal.
```

Comment:
Note: this is an example of a principality warning that was raised during exhaustivity analysis, on a sub-case of an "exploded" wildcard pattern. The dummy location (File "_none_", line 1) comes from the dummy location used by parmatch when exploding wildcard partterns.

I believe that the change in this testsuite result does not come from the lowering of level, but simply from the fact that we now silence this warning during exhaustivity analysis. (We silenced them because our behavior change would raise those warnings very often in this situation.) It's interesting that it already occurred in the wild.

Reply:
Indeed, it purely come from my turning off the warning in that.
Whether turning off the warning is correct might be debatable. Namely, the fact the inference is deemed not principal suggests that the exhaustiveness assessment might not be principal. However, at least the current warning was not satisfactory, as it warns on non-existing code. Moreover, the exhaustiveness check runs after everything else, so at that point all types should be already known, even if it is not in a principal way, so the surprise factor seems much lower.